### PR TITLE
[G2M]Diff/calendar

### PR DESCRIPTION
**Pull request by tlowande**
on top of #55 
Add scheduled demos from [demo calendar](https://calendar.google.com/calendar/u/0?cid=Y18wZGdoZ3MyNWo3cWplNmFhZmw0NDhybXQxY0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t) using google api
if no demos scheduled, there's no message

image

### CODE REVIEW

File: modules/diff.js [hunk 1]
```diff
@@ -115,6 +117,15 @@ const getGitDiff = async ({ product, base, head = 'master', dev, prod }) => {
   if (contributors.size > 1) {
     r.attachments[0].blocks[0].text.text += `\n\nContributors: ${Array.from(contributors).join(', ')}`
   }
+
+  // link to demos calendar
+  if (demos) {
+    const {day, link, events } = demos
```
Comment (woozyking) [inline modules/diff.js:123]: some style inconsistency

File: modules/diff.js [hunk 2]
```diff
@@ -115,6 +117,15 @@ const getGitDiff = async ({ product, base, head = 'master', dev, prod }) => {
...
+    r.attachments[0].blocks[0].text.text += `\n\n*Demos* on <${link}|${day}>`
+    events.forEach(({ timeSloth }) => {
+      r.attachments[0].blocks[0].text.text += `\n\t• ${timeSloth}`
```
Comment (woozyking) [inline modules/diff.js:126]: what is `•`?

Comment (tlowande) [reply modules/diff.js:126]: should work for mrkdwn, I know, but no  ... I remeber the blocks being more elaborated for list and list items... 🤷 
image

Comment (woozyking) [reply modules/diff.js:126]: :trollface:

**Review by woozyking — APPROVED** [review]

File: google-api/googleapis.js [hunk 3]
```diff
@@ -14,6 +14,57 @@ const oauth2Client = new google.auth.OAuth2(
...
+      htmlLink
+    },
+    i) => {
+    const eventKey = `event${i}`
+    const  existingEvents = Object.entries(acc || {})
+    let replaced = false
+    if (existingEvents.length) {
+      existingEvents.forEach(([key, { start, end }]) => {
+        const s = startDate >= start && startDate <= end
+        const e = endDate >= start && endDate <= end
```
Comment (tlowande) [inline google-api/googleapis.js:33]: not sure if the comparison between date strings `"2020-12-21T10:30:00-05:00"` (api response) are ok or if I should `new Date("2020-12-21T10:30:00-05:00").getTime()` here to play safe

Comment (woozyking) [reply google-api/googleapis.js:33]: if without the time zone then i'm 100% that'd be safe. with time zone not so much. perhaps use getTime for safety

**Pull request merged**